### PR TITLE
Remove duplicate s3 templates.

### LINKED ISSUE #357: Remove duplicate templates for S3
There are templates present in examples as well as in ocp-templates, kube-template.

It is far better to avoid this duplicity. 
let us remove the examples one. 

Also, document about deploying S3 containers using these new templates.

### PR DESCRIPTION
**Pull request by SaravanaStorageNetwork**
Update README to reflect new s3 template.
Fixes #357.

Signed-off-by: Saravanakumar <sarumuga@redhat.com>

---
This change is [Reviewable](https://reviewable.io/reviews/gluster/gluster-kubernetes/360)

### CODE REVIEW

File: docs/examples/gluster-s3-storage-template/README.md [hunk 1]
```diff
@@ -2,111 +2,132 @@
...
+OpenShift setup is up with master and nodes ready.
 
-  cns-deploy tool is ran and heketi service is ready
+cns-deploy tool is ran and heketi service is ready.
 
+## First create a storageclass
 
+We need to create a storageclass which provides access to glusterfs provisioner.
 
+Before creating storageclass, secret needs to be created and labelled. Please note, this step is optional.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Replace with:
```
In our example, we have set up heketi to require a secret key for the admin 
user. A StorageClass created to use such a heketi instance needs a Secret 
that contains the admin key. This Secret is not needed if heketi is not 
configured to use an admin key.
```

File: docs/examples/gluster-s3-storage-template/README.md [hunk 2]
```diff
@@ -2,111 +2,132 @@
...
 
-## First create a storageclass
+Create the secret:
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Remove this line

File: docs/examples/gluster-s3-storage-template/README.md [hunk 3]
```diff
@@ -2,111 +2,132 @@
...
 
-Create a GlusterFS StorageClass as below, replacing the `rest` parameters with your configuration:
+Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Explain what these parameters are.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 4]
```diff
@@ -2,111 +2,132 @@
...
-oc create -f ./gluster-s3-storageclass.yaml
+oc create secret generic heketi-${NAMESPACE}-admin-secret --from-literal=key=${ADMIN_KEY} --type=kubernetes.io/glusterfs
 ```
 
-Available at 
-[gluster-s3-storageclass.yaml](./gluster-s3-storageclass.yaml)
+Label the secret:
 
-## Start glusters3 service using template
+Replace `NAMESPACE` parameter with your configuration.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Rewrite it to something like:
```
As an optional step, the Secret can be labeled. This is useful to be able to 
select the secret as part of a general query like `oc get 
--selector=glusterfs` and allows the secret to be removed programatically by 
the `gk-deploy` tool.
```

File: docs/examples/gluster-s3-storage-template/README.md [hunk 5]
```diff
@@ -2,111 +2,132 @@
...
 
 ```
-oc new-app gluster-s3-template.yaml  --param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser --param=S3_PASSWORD=itsmine --param=VOLUME_CAPACITY=2Gi
+oc label --overwrite secret heketi-${NAMESPACE}-admin-secret glusterfs=s3-heketi-${NAMESPACE}-admin-secret gluster-s3=heketi-${NAMESPACE}-admin-secret
 ```
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Combine this sentence into previous sentence. Explain what all the new parameters are for (e.g. `HEKETI_URL` and `STORAGE_CLASS`) and where they'll be used later.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 6]
```diff
@@ -2,111 +2,132 @@
 
 ## Prerequisites
 
-  OpenShift setup is up with master and nodes ready.
+OpenShift setup is up with master and nodes ready.
 
-  cns-deploy tool is ran and heketi service is ready
+cns-deploy tool is ran and heketi service is ready.
 
+## First create a storageclass
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Insert a line above this one:
```
## Deployment
```
Then change this line to:
```
### 1. Provide the backend store
```

File: docs/examples/gluster-s3-storage-template/README.md [hunk 7]
```diff
@@ -2,111 +2,132 @@
 
 ## Prerequisites
 
-  OpenShift setup is up with master and nodes ready.
+OpenShift setup is up with master and nodes ready.
 
-  cns-deploy tool is ran and heketi service is ready
+cns-deploy tool is ran and heketi service is ready.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Change the above two lines to a bulleted list, and update the second line to `cns-deploy tool has been run`

File: docs/examples/gluster-s3-storage-template/README.md [hunk 8]
```diff
@@ -2,111 +2,132 @@
...
 
+## First create a storageclass
 
+We need to create a storageclass which provides access to glusterfs provisioner.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Elaborate on what is actually required:
```
The gluster-s3 service requires there be at least two GlusterFS volumes 
available for its use, one to store the object data and another for the 
object metadata. In this example, we will create a new StorageClass to 
dynamically provision these two volumes on our pre-existing GlusterFS cluster.

#### Create a StorageClass
```

File: docs/examples/gluster-s3-storage-template/README.md [hunk 9]
```diff
@@ -2,111 +2,132 @@
...
+Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
 
 ```
-oc create -f ./gluster-s3-storageclass.yaml
+oc create secret generic heketi-${NAMESPACE}-admin-secret --from-literal=key=${ADMIN_KEY} --type=kubernetes.io/glusterfs
 ```
 
-Available at 
-[gluster-s3-storageclass.yaml](./gluster-s3-storageclass.yaml)
+Label the secret:
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Remove this line

File: docs/examples/gluster-s3-storage-template/README.md [hunk 10]
```diff
@@ -2,111 +2,132 @@
...
+
+```
+sed  -e 's/${HEKETI_URL}/heketi-store-project1.cloudapps.mystorage.com/g'  -e 's/${STORAGE_CLASS}/gluster-s3-store/g'      -e  's/${NAMESPACE}/store-project1/g'   deploy/ocp-templates/gluster-s3-storageclass.yaml | oc create -f -
+```
 
+Available at 
+[gluster-s3-storageclass.yaml](../../../deploy/ocp-templates/gluster-s3-storageclass.yaml)
 
-If you wish to make use of a GlusterFS StorageClass other than `s3storageclass`, add another parameter of the form:
+## Create PVC
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Change to `#### Create backend PVCs`

File: docs/examples/gluster-s3-storage-template/README.md [hunk 11]
```diff
@@ -2,111 +2,132 @@
...
 
+Now, create PVC using the storage class. Replace `STORAGE_CLASS` with the above created one.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: `s/PVC/PVCs/` and `s/storage class/StorageClass`

File: docs/examples/gluster-s3-storage-template/README.md [hunk 12]
```diff
@@ -2,111 +2,132 @@
...
+Adjust `VOLUME_CAPACITY` as per your needs.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Combine with previous sentence.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 13]
```diff
@@ -2,111 +2,132 @@
...
 ```
---param=STORAGE_CLASS=<your storage class name>
+sed -e 's/${VOLUME_CAPACITY}/2Gi/g'  -e  's/${STORAGE_CLASS}/gluster-s3-store/g'  deploy/ocp-templates/gluster-s3-pvcs.yaml | oc create -f -
+persistentvolumeclaim "gluster-s3-claim" created
 ```
 
+Available at
+[gluster-s3-pvcs.yaml](../../../deploy/ocp-templates/gluster-s3-pvcs.yaml)
+
+## Start glusters3 service using template
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Change to `### 2. Start gluster-s3 service`

File: docs/examples/gluster-s3-storage-template/README.md [hunk 14]
```diff
@@ -2,111 +2,132 @@
...
 
+Launch S3 storage service.
+
+Set S3_ACCOUNT name, S3_USER name, S3_PASSWORD.
+PVC and META_PVC are obtained from above step.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Combine this with the previous sentence, and explain what the `S3_*` parameters are and that they can be whatever the admin wants to configure.

Also explain that PVC and META_PVC are the PVC names.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 15]
```diff
@@ -2,111 +2,132 @@
...
-If you wish to make use of a GlusterFS StorageClass other than `s3storageclass`, add another parameter of the form:
+## Create PVC
 
+Now, create PVC using the storage class. Replace `STORAGE_CLASS` with the above created one.
+Adjust `VOLUME_CAPACITY` as per your needs.
 
 ```
---param=STORAGE_CLASS=<your storage class name>
+sed -e 's/${VOLUME_CAPACITY}/2Gi/g'  -e  's/${STORAGE_CLASS}/gluster-s3-store/g'  deploy/ocp-templates/gluster-s3-pvcs.yaml | oc create -f -
+persistentvolumeclaim "gluster-s3-claim" created
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Include output for the meta claim.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 16]
```diff
@@ -2,111 +2,132 @@
...
+
+Set S3_ACCOUNT name, S3_USER name, S3_PASSWORD.
+PVC and META_PVC are obtained from above step.
+
+```
+oc new-app  deploy/ocp-templates/gluster-s3-template.yaml \
+--param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser \
+--param=S3_PASSWORD=itsmine --param=PVC=gluster-s3-claim \
+--param=META_PVC=gluster-s3-meta-claim
+```
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Remove this block, just use the one below.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 17]
```diff
@@ -2,111 +2,132 @@
...
 
 Available at:
-[gluster-s3-template.yaml](./gluster-s3-template.yaml)
+[gluster-s3-template.yaml](../../../deploy/ocp-templates/gluster-s3-template.yaml)
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Move these two lines below the following block of output.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 18]
```diff
@@ -2,111 +2,132 @@
...
 
 ### For example:
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Remove this line.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 19]
```diff
@@ -2,111 +2,132 @@
...
-    deploymentconfig "glusters3" created
+    service "gluster-s3-service" created
+    route "gluster-s3-route" created
+    deploymentconfig "gluster-s3-dc" created
 --> Success
     Run 'oc status' to view your app.
 ```
 
-
 ```
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Insert the following:
```
### 3. Verify gluster-s3 resources

Use the following commands to verify the deployment was succesful.
```

File: docs/examples/gluster-s3-storage-template/README.md (hunk 7 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 3 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 16 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 17 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 6 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 8 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 1 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 2 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 9 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 4 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 5 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Updated. check whether changes looks fine?

File: docs/examples/gluster-s3-storage-template/README.md (hunk 10 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 11 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 13 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 19 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md [hunk 20]
```diff
@@ -2,111 +2,136 @@
...
+dynamically provision these two volumes on our pre-existing GlusterFS cluster.
 
-## First create a storageclass
+#### Create a StorageClass
 
-Create a GlusterFS StorageClass as below, replacing the `rest` parameters with your configuration:
+In our example, we have set up heketi to require a secret key for the admin user. A StorageClass created to use such a heketi instance needs a Secret that contains the admin key. This Secret is not needed if heketi is not configured to use an admin key.
+
+ Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
+Here, `NAMESPACE` is the project and `ADMIN_KEY` is used for authorization to access Heketi service.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Remove the newline on the previous line.

File: docs/examples/gluster-s3-storage-template/README.md (hunk 5 above)
Comment (jarrpa) [reply docs/examples/gluster-s3-storage-template/README.md]: As they are currently written, all four of these single-sentence lines should be part of the same paragraph with line 38. You could also turn these four lines into a bulleted list if you prefer, but if so reqord the STORAGE_CLASS line to match the other three. :)

File: docs/examples/gluster-s3-storage-template/README.md [hunk 21]
```diff
@@ -2,111 +2,136 @@
...
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
+`HEKETI_URL` is the REST url to access GlusterFs cluster.
+`NAMESPACE` is the project.
+`STORAGE_CLASS` is the StorageClass name provided by admin.
 
+```
+sed  -e 's/${HEKETI_URL}/heketi-store-project1.cloudapps.mystorage.com/g'  -e 's/${STORAGE_CLASS}/gluster-s3-store/g'      -e  's/${NAMESPACE}/store-project1/g'   deploy/ocp-templates/gluster-s3-storageclass.yaml | oc create -f -
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Remove the extra spaces before the `-e` flags.

File: docs/examples/gluster-s3-storage-template/README.md [hunk 22]
```diff
@@ -2,111 +2,136 @@
...
---param=STORAGE_CLASS=<your storage class name>
+sed -e 's/${VOLUME_CAPACITY}/2Gi/g'  -e  's/${STORAGE_CLASS}/gluster-s3-store/g'  deploy/ocp-templates/gluster-s3-pvcs.yaml | oc create -f -
+persistentvolumeclaim "gluster-s3-claim" created
+persistentvolumeclaim "gluster-s3-meta-claim" created
 ```
 
+Available at
+[gluster-s3-pvcs.yaml](../../../deploy/ocp-templates/gluster-s3-pvcs.yaml)
 
+### 2. Start glusters3 service
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: `s/glusters3/gluster-s3/`

File: docs/examples/gluster-s3-storage-template/README.md [hunk 23]
```diff
@@ -2,111 +2,136 @@
...
 
-Available at:
-[gluster-s3-template.yaml](./gluster-s3-template.yaml)
+Launch S3 storage service.
+Set S3_ACCOUNT name, S3_USER name, S3_PASSWORD according to the user wish. S3_ACCOUNT is the S3 account which will be created and associated with GlusterFS volume. S3_USER is the user created to access the above account and S3_PASSWORD is for Authorization of the S3 user.  
+PVC and META_PVC are persistentvolumeclaim(s) obtained from above step.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Combine this line with the previous line, and enclose all VARIABLE names with backticks (`` ` ``).

File: docs/examples/gluster-s3-storage-template/README.md [hunk 24]
```diff
@@ -2,111 +2,136 @@
...
-NAME              HOST/PORT                                                            PATH      SERVICES           PORT      TERMINATION   WILDCARD
-glusters3object   glusters3object-storage-project.cloudapps.mystorage.com ... 1 more             glusters3service   <all>                   None
-heketi            heketi-storage-project.cloudapps.mystorage.com ... 1 more                      heketi             <all>                   None
+# oc get route gluster-s3-route
+NAME               HOST/PORT                                                             PATH      SERVICES             PORT      TERMINATION   WILDCARD
+gluster-s3-route   gluster-s3-route-storage-project.cloudapps.mystorage.com ... 1 more             gluster-s3-service   <all>                   None
+
 ```
+Available at:
+[gluster-s3-template.yaml](../../../deploy/ocp-templates/gluster-s3-template.yaml)
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: Move these two likes above `### 3. Verify gluster-s3 resources`

File: docs/examples/gluster-s3-storage-template/README.md [hunk 25]
```diff
@@ -2,111 +2,136 @@
...
 
 ```
-oc new-app gluster-s3-template.yaml  --param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser --param=S3_PASSWORD=itsmine --param=VOLUME_CAPACITY=2Gi
+oc label --overwrite secret heketi-${NAMESPACE}-admin-secret glusterfs=s3-heketi-${NAMESPACE}-admin-secret gluster-s3=heketi-${NAMESPACE}-admin-secret
 ```
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
+`HEKETI_URL` is the REST url to access GlusterFs cluster.
```
Comment (jarrpa) [inline docs/examples/gluster-s3-storage-template/README.md]: `s/REST url/URL/`, `s/GlusterFs/GlusterFS/`

File: docs/examples/gluster-s3-storage-template/README.md (hunk 20 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 25 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 21 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 22 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 23 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

File: docs/examples/gluster-s3-storage-template/README.md (hunk 24 above)
Comment (SaravanaStorageNetwork) [reply docs/examples/gluster-s3-storage-template/README.md]: Done

**Pull request merged**